OSDOCS-5759 Removal of kube-controller-manager nodelifecycle controller's fields (#65336)
openshift/openshift-docs: modules/nodes-edge-remote-workers-strategies.adoc

<3> Specify the frequency that the kubelet reports the status of a node associated with this `MachineConfig` object. The default value is `1m`.

The `node-status-update-frequency` parameter works with the `node-monitor-grace-period` parameter.

* The `node-monitor-grace-period` parameter specifies how long {product-title} waits after a node associated with a `MachineConfig` object is marked `Unhealthy` if the controller manager does not receive the node heartbeat. Workloads on the node continue to run after this time. If the remote worker node rejoins the cluster after `node-monitor-grace-period` expires, pods continue to run. New pods can be scheduled to that node. The `node-monitor-grace-period` interval is `40s`. The `node-status-update-frequency` value must be lower than the `node-monitor-grace-period` value.

[NOTE]
====
Modifying the `node-monitor-grace-period` parameter is not supported.
====

--

[id="nodes-edge-remote-workers-strategies-tolerations_{context}"]
Tolerations::

> You can use pod tolerations to mitigate the effects if the on-premise node controller adds a `node.kubernetes.io/unreachable` taint with a `NoExecute` effect to a node it cannot reach. 

Reviewer: extra space at the end?

A taint with the `NoExecute` effect affects pods that are running on the node in the following ways:

* Pods that do not tolerate the taint are queued for eviction.
* Pods that tolerate the taint without specifying a `tolerationSeconds` value in their toleration specification remain bound forever.

> * Pods that tolerate the taint with a specified `tolerationSeconds` value remain bound for the specified amount of time. After the time elapses, the pods are queued for eviction. Kubernetes sets the default `tolerationSeconds``, however you can set a preferred value. 

Reviewer: It does not just set the tolerationSeconds. It adds full `node.kubernetes.io/not-ready` and `node.kubernetes.io/unreachable` tolerations by default. See `Note` sections in https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-based-evictions

With this in mind, it is probably better to mention the defaulting in the first point or make another point out of it?

* Pods that do not tolerate the taint are queued for eviction.
* Pods that tolerate the taint without specifying a `tolerationSeconds` value in their toleration specification remain bound forever.
* Pods that tolerate the taint with a specified `tolerationSeconds` value remain bound for the specified amount of time. After the time elapses, the pods are queued for eviction.

[NOTE]
====

> Unless tolerations are explicitly set, Kubernetes will automatically add a toleration for `node.kubernetes.io/not-ready` and `node.kubernetes.io/unreachable` with `tolerationSeconds=300`, meaning that pods will remain bound for 5 minutes if either of these taints is detected. 

Reviewer 2: ```suggestion
Unless tolerations are explicitly set, Kubernetes automatically adds a toleration for `node.kubernetes.io/not-ready` and `node.kubernetes.io/unreachable` with `tolerationSeconds=300`, meaning that pods remain bound for 5 minutes if either of these taints is detected. 
```

Reviewer 2: Avoid using future tense:
https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs#tense

Author: Fixed, thanks for the catch!

> Unless tolerations are explicitly set, Kubernetes will automatically add a toleration for `node.kubernetes.io/not-ready` and `node.kubernetes.io/unreachable` with `tolerationSeconds=300`, meaning that pods will remain bound for 5 minutes if either of these problems is detected. 

Reviewer: +1, looks better

```suggestion
Unless tolerations are explicitly set, Kubernetes will automatically add a toleration for `node.kubernetes.io/not-ready` and `node.kubernetes.io/unreachable` with `tolerationSeconds=300`, meaning that pods will remain bound for 5 minutes if either of these taints is detected. 
```

Author: Fixed, thanks!

- key: "node.kubernetes.io/not-ready"
  operator: "Exists"
  effect: "NoExecute" <2>

>   tolerationSeconds: "300" <3>

Reviewer: probably better to keep the old value or have a different one, because 300s (5m) is the default behaviour for this particular toleration

Author: Is the default 5 minutes or no limit? Or, is it that if the `tolerationSeconds` variable is not applied then the default is no limit?

Reviewer: if there is no toleration entry with one of these two keys, then a new entry will be added with 5 minutes long `tolerationSeconds` for each missing key

- key: "node.kubernetes.io/not-ready"
  operator: "Exists"
  effect: "NoExecute" <2>
  tolerationSeconds: 600 <3>
...
----
<1> The `NoExecute` effect without `tolerationSeconds` lets pods remain forever if the control plane cannot reach the node.
<2> The `NoExecute` effect with `tolerationSeconds`: 600 lets pods remain for 10 minutes if the control plane marks the node as `Unhealthy`.
<3> You can specify your own `tolerationSeconds` value.

> Other types of {product-title} objects::

Reviewer 3: ```suggestion

Other types of {product-title} objects::
```
This is previewing as part of the footnote, but it looks like it ought to be a glossary-style section.

Author: I think I fixed it, wdyt?

Reviewer 3: Looks good to me!

- key: "node.kubernetes.io/not-ready"
  operator: "Exists"
  effect: "NoExecute" <2>
  tolerationSeconds: 600 <3>
...
----
<1> The `NoExecute` effect without `tolerationSeconds` lets pods remain forever if the control plane cannot reach the node.

> <2> The `NoExecute` effect with `tolerationSeconds`: 300 lets pods remain for 5 minutes if the control plane marks the node as `Unhealthy`.

Reviewer: please update the times here as well
